File: ExampleObjc/ExampleObjc/ViewController.m
//
//  ViewController.m
//  ExampleObjc
//
//  Created by Rodrigo Busata on 05/12/21.
//

#import "ViewController.h"
#import <SmartPromo/SmartPromo.h>

@interface ViewController ()<SmartPromoDelegate>

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
}

- (IBAction)actionOpen:(id)sender {
    SmartPromo* smartPromo = [SmartPromo new];
    smartPromo.delegate = self;
    
    [smartPromo setupAccessKey:@"{accessKey}"
                  andSecretKey:@"{secretKey}"];
    
    FSPConsumer* consumer = [FSPConsumer new];
    consumer.cpf = @"{cpf}";
    [smartPromo setConsumer:consumer];
    
    [smartPromo go:@"{campaignID}" above: self];
}

- (void)smartPromoDidReceiveEvent:(NSString *)eventKey values:(NSDictionary<NSString *,id> *)values {
    NSLog(@"[SmartPromo] Event received: %@", eventKey);
    
    if (values.count > 0) {
        NSLog(@"[SmartPromo] Event values: %@", values);
    } else {
        NSLog(@"[SmartPromo] No values attached to event");
    }
}

@end

File: Example/Example/ViewController.swift
//
//  ViewController.swift
//  Example
//
//  Created by Rodrigo Busata on 05/10/21.
//

import UIKit
import SmartPromo

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
    }

    @IBAction func actionOpen(_ sender: Any) {
        let smartPromo = SmartPromo()
        smartPromo.delegate = self
        
        smartPromo.setupAccessKey("{accessKey}", andSecretKey: "{secretKey}")
        
        let consumer = FSPConsumer()
        consumer.cpf = "{cpf}"
        smartPromo.setConsumer(consumer)
        smartPromo.setIsHomolog(true)
        
        smartPromo.go("{campaignID}", above: self)
    }
}

extension ViewController: SmartPromoDelegate {
    
    func smartPromoDidReceiveEvent(_ eventKey: String, values: [String : Any]?) {
        print("[SmartPromo] Event received: \(eventKey)");
            
        if let values {
            print("[SmartPromo] Event values: \(values)");
        } else {
            print("[SmartPromo] No values attached to event");
        }
    }
}
